clustering/src/main/java/io/hyperfoil/controller/Data.java

```diff
@@ -27,26 +28,30 @@ final class Data {
    final int stepId;
    final String metric;
    // for reporting
+   private final Boolean trackIntervalHistograms;
    final StatisticsSnapshot total = new StatisticsSnapshot();
    final Map<String, StatisticsSnapshot> perAgent = new HashMap<>();
    final Map<String, IntObjectMap<StatisticsSnapshot>> lastStats = new HashMap<>();
    final List<StatisticsSummary> series = new ArrayList<>();
+   final Map<String, List<Histogram>> intervalHistogramsPerAgent = new HashMap<>();;
+   final List<Histogram> intervalHistograms = new ArrayList<>();
    final Map<String, List<StatisticsSummary>> agentSeries = new HashMap<>();
    // floating statistics for SLAs
    private final Map<SLA, StatisticsStore.Window> windowSlas;
    private final SLA[] totalSlas;
-   private int highestSequenceId = 0;
+   private int highestSampleId = 0;
    private boolean completed;
 
    Data(StatisticsStore statisticsStore, String phase, boolean isWarmup, int stepId, String metric,
-         Map<SLA, StatisticsStore.Window> periodSlas, SLA[] totalSlas) {
+         Map<SLA, StatisticsStore.Window> periodSlas, SLA[] totalSlas, Boolean trackIntervalHistograms) {
       this.statisticsStore = statisticsStore;
       this.phase = phase;
       this.isWarmup = isWarmup;
       this.stepId = stepId;
       this.metric = metric;
       this.windowSlas = periodSlas;
       this.totalSlas = totalSlas;
+      this.trackIntervalHistograms = trackIntervalHistograms;
```

Comment:
With regards to Franz's comment on not allocating if we're not saving we can add a check here if `trackIntervalHistograms == true` and instantiate `intervalHistogramsPerAgent` and `intervalHistograms` only if it is?

Reply:
yeah that's something I can add as well, at the moment they are always instantiated but kept empty if `trackIntervalHistograms == true`